core/src/main/protobuf/org/apache/spark/status/protobuf/store_types.proto

```diff
@@ -22,7 +22,12 @@ package org.apache.spark.status.protobuf;
  * Developer guides:
  *   - Coding style: https://developers.google.com/protocol-buffers/docs/style
  *   - Use int64 for job/stage IDs, in case of future extension in Spark core.
- *   - Use `weakIntern` on string values in create new objects during deserialization.
+ *   - For string fields:
+ *     - use `optional string` for protobuf definition
+ *     - on serialization, check if the string is null to avoid NPE
+ *     - on deserialization, set string fields as null if it is not set. Also, use `weakIntern` on
+ *       string values in create new objects during deserialization.
+ *     - add tests with null string inputs
```

Comment:
Just a question. Is there a way to automate this in `store_types.proto` file?

Reply:
AFAIK no. Unless we modify the protoc for generating java. I think having tests for setting all string fields as nulls should be enough for now.